test/lint/git-subtree-check.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017-2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Checks that a certain prefix is pure subtree, and optionally whether the
+# referenced commit is present in any fetched remote.
+
+import argparse
+import sys
+import subprocess
+
+def parse_args():
+    """Parse command line arguments"""
+    parser = argparse.ArgumentParser (
+        description="""
+            Checks that a certain prefix is pure subtree, and optionally whether the
+            referenced commit is present in any fetched remote.
+        """
+    )
+
+    parser.add_argument('dir', help="The prefix within the repository to check.")
+    parser.add_argument('commit', nargs='?', default='HEAD', help="The commit to check, if it is not provided, HEAD will be used.")
+    parser.add_argument("--repository", "-r", action='store_true', required=False, help="Check that subtree commit is present in repository. To do this check, fetch the subtreed remote first. (e.g. git fetch https://github.com/bitcoin-core/secp256k1.git\ntest/lint/git-subtree-check.sh -r src/secp256k1)")
+
+    return parser.parse_args()
+
+def find_latest_squash(dir, commit):
+    sq = main = sub = None
+    stdout = subprocess.check_output(["git", "log", "--grep=^git-subtree-dir: " + dir + "/*$", "--pretty=format: START %H%n%s%n%n%b%nEND%n", commit], universal_newlines=True, encoding="utf8")
+    for line in stdout.splitlines():
+        if "START" in line:
+            sq = line.lstrip().split(' ')[1]
+        if "git-subtree-mainline:" in line:
+            main = line.split(' ')[1]
+        if "git-subtree-split:" in line:
+            sub = line.split(' ')[1]
+        if "END" in line:
+            if sub:
+                if main:
+                    sq = sub
+                return(f"{sq} {sub}")
+            sq = main = sub = None
+
+def main():
+    args = parse_args()
+
+    # find the latest subtree update
+    latest_squash = find_latest_squash(args.dir.rstrip('/'), args.commit)
+    if not latest_squash:
+        print(f"Error: {args.dir} is not a subtree")
+        sys.exit(2)
+    else:
+        old = latest_squash.split(' ')[0]
+        rev = latest_squash.split(' ')[1]
+
+    # get the tree in the current commit
+    tree_actual = subprocess.check_output(["git", "ls-tree", "-d", args.commit, args.dir, "| head -n 1"], universal_newlines=True, encoding="utf8")
```

Comment:
whoops: `| head -n 1` is not going to work here, please implement this in Python

Reply:
fixed by piping output to new subprocess